### PR TITLE
Replace deprecated Rollup option 'legacy' with custom plugin.

### PR DESCRIPTION
**Pull request by elas7**
The `legacy` option was added to the Rollup build system (for www builds) in #11469 because the internal transforms break on getters. As of version 0.60.0 of Rollup, `legacy` support is removed ([source](https://github.com/rollup/rollup/pull/2141), [changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md#0600)). This PR adds a custom Rollup plugin to replicate the `legacy` behaviour.

Note that the `legacy` option performed several transformations to support IE8. The custom Rollup plugin only replicates the replacement of getters ([original code here](https://github.com/rollup/rollup/blob/349677ceee9d4bbccb5b2f72e653270cb2b0ce51/src/ast/variables/NamespaceVariable.ts#L63)).

Also note that getters are only generated in a very specific case by Rollup. That is, when doing namespace imports (`import * as name from 'origin'`) and one of the imported variables is reassigned at some point (an example of this can be found in the [old Rollup tests](https://github.com/rollup/rollup/tree/349677ceee9d4bbccb5b2f72e653270cb2b0ce51/test/form/samples/legacy-getter)). At this time, the only getter generated by React code is `ReactDOMEventListener._enabled`.

### CODE REVIEW

**Comment by gaearon** [comment]
Seems brittle.
Can we run a final Babel pass instead on the final bundle to just transform the getters?

**Comment by TrySound** [comment]
@gaearon Is this still relevant for fb build? In https://github.com/facebook/react/pull/13321 I just removed this option.

**Comment by elas7** [comment]
@gaearon Currently Babel runs before Rollup, but I'll try to add a new Babel pass after the Rollup bundle to remove the getters.

**Comment by gaearon** [comment]
Yeah. Just for FB bundle, and just with that single transform.

**Comment by gaearon** [comment]
Is there a reason we're not using some stock Babel transform for compiling getters according to the spec?

**Comment by elas7** [comment]
@gaearon because there is no stock Babel plugin to convert getters to ES3 ([source](https://babeljs.io/docs/en/plugins#es3))

File: scripts/rollup/build.js [hunk 1]
```diff
@@ -498,6 +494,24 @@ async function createBundle(bundle, bundleType) {
...
+      babelrc: false,
+      compact: false,
+      plugins: [
+        // This is needed because Rollup outputs getters, but we can't use
+        // getters in www.
+        require('../babel/remove-getters'),
+      ],
+    };
+    const result = babelCore.transform(bundleCode, babelOptions);
+    fs.writeFileSync(mainOutputPath, result.code, 'utf8');
```
[the remark this answers is not in the document]
Comment (elas7) [reply scripts/rollup/build.js]: @TrySound sounds good to me. I can make that change.

@gaearon are you ok with that?

**Comment by elas7** [comment]
@Simek Sadly I can't use that unofficial plugin because it only transforms `Object.defineProperty()` getters, and those are not the kind of getters generated by Rollup.

Thanks, though.

File: scripts/rollup/build.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (elas7) [reply scripts/rollup/build.js]: @TrySound done

File: scripts/babel/remove-getters.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        //
+        // prop: variable;
+        if (path.node.kind !== 'get') {
+          return;
+        }
+
+        const keyNode = path.node.key;
+        const isValidKey = t.isIdentifier(keyNode);
+        if (!isValidKey) {
+          return;
```
Comment (gaearon) [inline scripts/babel/remove-getters.js]: Can we throw here instead?

File: scripts/babel/remove-getters.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        }
+
+        const bodyNode = path.node.body;
+        const isValidBody =
+          bodyNode.body.length === 1 &&
+          t.isReturnStatement(bodyNode.body[0]) &&
+          t.isIdentifier(bodyNode.body[0].argument);
+        if (!isValidBody) {
+          return;
```
Comment (gaearon) [inline scripts/babel/remove-getters.js]: Here too.

**Pull request closed without merging**